common/failure/failure.go

```diff
@@ -52,8 +58,10 @@ func NewResetWorkflowFailure(message string, lastHeartbeatDetails *commonpb.Payl
 	return f
 }
 
-func NewTimeoutFailure(timeoutType enumspb.TimeoutType) *failurepb.Failure {
+func NewTimeoutFailure(message string, timeoutType enumspb.TimeoutType) *failurepb.Failure {
 	f := &failurepb.Failure{
+		Message: fmt.Sprintf("%v: %v", enumspb.TimeoutType_name[int32(timeoutType)], message),
```

Comment:
You need to swap arguments here:
```suggestion
		Message: fmt.Sprintf("%s: %v", message, timeoutType.String()]),
```